File: 000001-000200/000069.Sqrt(x)/69.go
package main

// 方法二: 用时 0ms
// 执行用时超过了 100%
// 消耗内存超过了 100%
func mySqrt(x int) int {
  if x == 0 {
    return 0
  }
  ret := int(math.Exp(0.5 * math.Log(float64(x))))
  if (ret + 1) * (ret + 1) <= x {
    return (ret + 1)
  }
  return ret
}

func main () {

}

File: 000001-000200/000069.Sqrt(x)/69.js
/**
 * @param {number} x
 * @return {number}
 */
 // 方法一: 用时 80ms
 // 执行用时超过了 86%
 // 消耗内存超过了 21%
var mySqrt = function(x) {
  if (x < 2) {
    return x
  }
  let left = 1
  let right = Math.floor(x / 2)
  while (left <= right) {
    let mid = Math.floor(left + (right - left) / 2)
    if (mid * mid === x) {
      return mid
    }
    if (mid * mid < x) {
      left = mid + 1
    } else {
      right = mid - 1
    }
  }
  return right
};

// 方法二: 用时 80ms
// 执行用时超过了 86%
// 消耗内存超过了 21%
var mySqrt = function (x) {

}

console.log(mySqrt(6))
console.log(mySqrt(10))
console.log(mySqrt(16))
